internal/pkg/otel/components.go

```diff
@@ -122,15 +121,18 @@ func components(extensionFactories ...extension.Factory) func() (otelcol.Factori
 		}
 
 		// Exporters
-		factories.Exporters, err = exporter.MakeFactoryMap(
+		exporters := []exporter.Factory{
 			otlpexporter.NewFactory(),
 			debugexporter.NewFactory(),
 			fileexporter.NewFactory(),
 			elasticsearchexporter.NewFactory(),
 			loadbalancingexporter.NewFactory(),
 			otlphttpexporter.NewFactory(),
-			kafkaexporter.NewFactory(),
-		)
+		}
+		// some exporters should only be available when
+		// not in fips mode due to restrictions on crypto usage
+		exporters = addNonFipsExporters(exporters)
+		factories.Exporters, err = exporter.MakeFactoryMap(exporters...)
```

Comment:
The prometheus and kafkareceivers are pretty functionally important. Is it even worth doing this vs just documenting they are potentially not compliant under some configurations? Or are they always non compliant?

We will need to figure out how to make these FIPS compliant eventually, and hopefully be able to push changes upstream. 

I'm not strictly opposed to this I just want to note it is kicking the can down the road.

Reply:
How valuable is the otel mode when disabling kafka and prometheus receivers right now? What we have discussed was to potentially completely disable otel when using fips for now, and then have a dedicated workstream (outside of current scope) for fixing problematic libriaries in otel mode as this might be a larger undertaking.

Reply:
Discussed offline that we want to keep anything from otel that is not problematic enabled and create follow up cards (https://github.com/elastic/elastic-agent-libs/issues/300) for fixing anything that is disabled here.